sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2248,21 +2244,37 @@ public <T> Mono<FeedResponse<T>> readMany(
                         // partitionkeyrange
                         Map<PartitionKeyRange, SqlQuerySpec> rangeQueryMap;
                         rangeQueryMap = getRangeQueryMap(partitionRangeItemKeyMap,
+                            singleItemPartitionRequestMap,
                             collection.getPartitionKey());
 
+                        // create point reads
+                        Mono<? extends List<FeedResponse<Document>>> pointReads = singleItemPartitionRequestMap.values().size() > 0 ? createPointReadOperations(
+                            singleItemPartitionRequestMap,
+                            resourceLink,
+                            options,
+                            klass)
+                            .collectList() : Mono.just(new ArrayList<>());
+
+
                         // create the executable query
-                        return createReadManyQuery(
+                        Mono<? extends List<FeedResponse<Document>>> queries = rangeQueryMap.keySet().size() > 0 ? createReadManyQuery(
                             resourceLink,
                             new SqlQuerySpec(DUMMY_SQL_QUERY),
                             options,
                             Document.class,
                             ResourceType.Document,
                             collection,
                             Collections.unmodifiableMap(rangeQueryMap))
-                            .collectList() // aggregating the result construct a FeedResponse and
-                            // aggregate RUs.
+                            .collectList() : Mono.just(new ArrayList<>());
+
+                        // merge results from point reads and queries
+                        return Flux.merge(pointReads, queries)
+                            .flatMapIterable(feedResponses -> feedResponses)
```

Comment:
Based on above suggestions on directly using Flux<FeedResponse<Document>>, you probably don't need flatMapIterable operator here -> 

```
// merge results from point reads and queries
                        return Flux.merge(pointReads, queries)
                            .collectList()
                            // aggregating the result to construct a FeedResponse and aggregate RUs.
                            .map(feedList -> {
```